_posts/2020-09-16-creating-a-snap-chat-mini.markdown

```diff
@@ -0,0 +1,95 @@
+---
+layout: post
+title: "What we Learned Developing Snapchat Minis (Part 1)"
+date: 2020-09-16 12:00:00
+categories: ["learning"]
+author: arieljuod
+---
+
+Over the last few months, we developed a couple of (Snapchat Minis)[https://www.ombulabs.com/blog/software-development/snap-minis.html]. Minis are small static web apps that are run inside a webview within the Snapchat native app.
+
+One important part of the development was to make sure our mini works well across different devices, especially making sure things work the same in both Android's webview (Chrome by default) and iOS's webview (Safari).
+
+In this post I'll talk about a few things to keep in mind when working with the Span canvas SDK and in the next one I'll talk about some common and known issues you'll encounter when making your app cross-mobile-browser compatible.
+
+<!--more-->
+
+## Working with the SDK
+
+Snapchat provides a JavaScript SDK that you'll have to include in your project. This SDK works as a bridge between the native app and your web app and lets you trigger native functionalities (like sharing on chat and camera), it gives you information about the current user and session, and also sends different events so you can react to them (you can detect if a user takes a screenshot or changes the volume for example).
+
+Along with the SDK, Snapchat will provide guides, documentation, a manual and a support website with FAQs and useful information.
+
+### Initializing the SDK
+
+Along with the SDK files, you'll have the documentation and the manual to help you set up the SDK initialization. The process is simple and it's explained in the manual so we won't go over that here. However, one thing to keep in mind is is that the initialization callback won't be triggered outside the native app when you are developing the app on a desktop/laptop computer. If your app depends on the SDK being initialized to work properly, you'll have to implement a fallback mechanism to trigger the same callback locally.
+
+There are many ways to do so: you can use node or webpack to have different code executed in different environments, you can have a boolean flag that you flip manually before compiling the mini, etc. You can do what works best for your process.
+
+> We recommend always waiting for the SDK initialization before rendering the app to prevent issues.
+
+### Fallback Method Calls when Working Locally
+
+When developing locally, since the SDK won't be connected to a native app, you'll have to handle some fallbacks in the places where you need to call SDK methods (when sharing to camera, chat, etc for example). If you want to test the app flow more easily instead of publishing every change to access the mini using your phone.
```

Comment:
```suggestion
When developing locally, since the SDK won't be connected to a native app, you'll have to handle some fallbacks in the places where you need to call SDK methods (when sharing to camera or chat, for example) if you want to test the app flow more easily, instead of publishing every change so you can access the Mini using your phone.
```